test/e2e/tests/test_bucket.py

```diff
@@ -35,35 +34,39 @@
 def s3_client():
     return boto3.client("s3")
 
+@pytest.fixture(scope="module")
+def s3_resource():
+    return boto3.resource("s3")
+
 @service_marker
 @pytest.mark.canary
 class TestBucket:
-    def get_bucket(self, s3_client, bucket_name: str) -> dict:
+    def get_bucket(self, s3_resource, bucket_name: str):
+        return s3_resource.Bucket(bucket_name)
+
+    def bucket_exists(self, s3_client, bucket_name: str) -> bool:
         try:
             resp = s3_client.list_buckets()
         except Exception as e:
             logging.debug(e)
-            return None
+            return False
 
         buckets = resp["Buckets"]
         for bucket in buckets:
             if bucket["Name"] == bucket_name:
-                return bucket
-
-        return None
+                return True
 
-    def bucket_exists(self, s3_client, bucket_name: str) -> bool:
-        return self.get_bucket(s3_client, bucket_name) is not None
+        return False
 
-    def test_smoke(self, s3_client):
+    def _create_smoke(self, s3_client, resource_file_name: str):
```

Comment:
A [smoke test](https://en.wikipedia.org/wiki/Smoke_testing_(software)) is generally a (usually preliminary or fail-fast) test of very basic functionality that, should failure occur in one of the smoke test steps, it would be a warning that something is definitely amiss about the code under test.

The reason I'd named this test method `test_smoke` was just that: it was a stupid simple create-then-delete a bucket CR and check it was created/deleted in the S3 API.

What you've done in this PR is add additional test assertions around the new Bucket attributes, which is awesome, however there isn't really a "smoke test" any more :) I'd recommend just renaming this from `_create_smoke` to `_create_bucket` (and potentially just placing it as a module-level function instead of a method on the `TestBucket` class.

Reply:
Ahh yep. Okee dokee